File: backend/src/main/java/com/bookloop/api/notification/Notification.java
package com.bookloop.api.notification;

import com.bookloop.api.user.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private NotificationType type;

    @Column(nullable = false, length = 255)
    private String title;

    @Column(nullable = false, length = 500)
    private String message;

    @Column(nullable = false, name = "is_read")
    private Boolean read = false;

    @Column(nullable = false)
    private LocalDateTime createdAt;

}

File: mobile/models/Notification.ts
export interface Notification {
  id: number;
  title: string;
  message: string;
  type:
    | "SOLICITACAO_RECEBIDA"
    | "SOLICITACAO_ACEITA"
    | "MATCH_CRIADO"
    | "TRANSACAO_FINALIZADA";
  read: boolean;
  createdAt: string;
}